e2e/scenario_test.go

```diff
@@ -0,0 +1,937 @@
+	t.Run("ubuntu2204-artifact-streaming", Scenario_ubuntu2204ArtifactStreaming)
+	t.Run("ubuntu2204-chrony-restarts", Scenario_ubuntu2204ChronyRestarts)
+	t.Run("ubuntu2204-containerd-url", Scenario_ubuntu2204ContainerdURL)
+	t.Run("ubuntu2204-containerd-version", Scenario_ubuntu2204ContainerdVersion)
+	t.Run("ubuntu2204-custom-ca-trust", Scenario_ubuntu2204CustomCATrust)
+	t.Run("ubuntu2204-custom-sysctls", Scenario_ubuntu2204CustomSysctls)
+	t.Run("ubuntu2204-gpu-a10", Scenario_ubuntu2204gpua10)
+	t.Run("ubuntu2204-gpu-a100", Scenario_ubuntu2204gpua100)
+	t.Run("ubuntu2204-gpu-grid-driver", Scenario_ubuntu2204GPUGridDriver)
+	t.Run("ubuntu2204-gpu-ncv", Scenario_ubuntu2204gpuncv)
+	t.Run("ubuntu2204-gpu-no-driver", Scenario_ubuntu2204gpuNoDriver)
+	t.Run("ubuntu2204-private-kube-pkg", Scenario_ubuntu2204privatekubepkg)
+	t.Run("ubuntu2204-wasm", Scenario_ubuntu2204Wasm)
+}
+
+func Scenario_azurelinuxv2(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using a AzureLinuxV2 (CgroupV2) VHD can be properly bootstrapped",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDAzureLinuxV2Gen2,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-azurelinux-v2-gen2"
+				nbc.AgentPoolProfile.Distro = "aks-azurelinux-v2-gen2"
+			},
+		},
+	})
+}
+
+func Scenario_azurelinuxv2AirGap(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using a AzureLinuxV2 (CgroupV2) VHD can be properly bootstrapped",
+		Tags: Tags{
+			Airgap: true,
+		},
+		Config: Config{
+			Cluster: ClusterKubenetAirgap,
+			VHD:     config.VHDAzureLinuxV2Gen2,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-azurelinux-v2-gen2"
+				nbc.AgentPoolProfile.Distro = "aks-azurelinux-v2-gen2"
+			},
+		},
+	})
+}
+
+func Scenario_azurelinuxv2ARM64(t *testing.T) {
+	RunScenario(t, &Scenario{
+		Description: "Tests that a node using a AzureLinuxV2 (CgroupV2) VHD on ARM64 architecture can be properly bootstrapped",
+		Config: Config{
+			Cluster: ClusterKubenet,
+			VHD:     config.VHDAzureLinuxV2Gen2Arm64,
+			BootstrapConfigMutator: func(nbc *datamodel.NodeBootstrappingConfiguration) {
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].VMSize = "Standard_D2pds_V5"
+				nbc.ContainerService.Properties.AgentPoolProfiles[0].Distro = "aks-azurelinux-v2-arm64-gen2"
+				nbc.ContainerService.Properties.OrchestratorProfile.KubernetesConfig.CustomKubeBinaryURL = "https://acs-mirror.azureedge.net/kubernetes/v1.24.9/binaries/kubernetes-node-linux-arm64.tar.gz"
+				nbc.AgentPoolProfile.VMSize = "Standard_D2pds_V5"
+				nbc.AgentPoolProfile.Distro = "aks-azurelinux-v2-arm64-gen2"
+				nbc.IsARM64 = true
+			},
+			VMConfigMutator: func(vmss *armcompute.VirtualMachineScaleSet) {
```

Comment:
maybe in a follow-up you can rename `VMConfigMutator` to `VMSSModelMutator` to make it a bit more clear?